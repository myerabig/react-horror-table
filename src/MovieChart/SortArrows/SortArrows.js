import React from 'react';
import sortMovies from '../utils/sortMovies';

function SortArrows(props) {
  return (
    <div className='SortArrows'>
      <div align='right'>
        <img
          className='icon'
          id='upArrow'
          src={require('../../assets/uparrow.png')}
          width='20'
          alt='Up'
          onClick={() =>
            sortMovies(
              props.movies,
              props.setMovies,
              'desc',
              props.property,
              props.property2
            )
          }
        />
        <img
          className='icon'
          id='downArrow'
          src={require('../../assets/downarrow.png')}
          width='20'
          alt='Down'
          onClick={() =>
            sortMovies(
              props.movies,
              props.setMovies,
              'asc',
              props.property,
              props.property2
            )
          }
        />
      </div>
    </div>
  );
}

export default SortArrows;
