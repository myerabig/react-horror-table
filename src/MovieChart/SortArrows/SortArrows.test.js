import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import SortArrows from './SortArrows';

describe('SortArrows', () => {
  afterEach(cleanup);

  let MOCK_MOVIES = [
    {
      title: 'b',
      year: '2'
    },
    {
      title: 'a',
      year: '1'
    }
  ];
  const MOCK_SETMOVIES = movies => (MOCK_MOVIES = movies);

  const { container, getByAltText } = render(
    <SortArrows
      movies={MOCK_MOVIES}
      setMovies={MOCK_SETMOVIES}
      property='title'
      property2='year'
    />
  );

  it('should render as expected', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
  it.skip('should call sortMovies with "asc" on down arrow click', () => {
    var sortMovies = require('../utils/sortMovies');
    sortMovies = jest.fn();

    fireEvent(getByAltText('Up'), new MouseEvent('click'));

    expect(sortMovies).toHaveBeenCalled();
  });
});
