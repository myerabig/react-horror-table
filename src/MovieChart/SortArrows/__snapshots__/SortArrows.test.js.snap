// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortArrows should render as expected 1`] = `
<div
  class="SortArrows"
>
  <div
    align="right"
  >
    <img
      alt="Up"
      class="icon"
      id="upArrow"
      src="uparrow.png"
      width="20"
    />
    <img
      alt="Down"
      class="icon"
      id="downArrow"
      src="downarrow.png"
      width="20"
    />
  </div>
</div>
`;
